### ISSUE TITLE
Support http and grpc transcoding as AIP-127

### ISSUE THREAD

**Issue by devkanro** [timostamm/protobuf-ts#101]
In my usage case, we use the [AIP-127](https://aip.bybutter.com/127) to provide gRPC implementation on the web platform which doesn't support HTTP/2 completely.

I have built a [similar library](https://github.com/ButterCam/sisyphus-js) for our front-end team, but I am focusing on Backend and Java/Kotlin. As you see, the library has no update for a long time, it only just works, not a high-performance choice.

Maybe we can switch to the `protobuf-ts` for our front-end team. Before we start trying it, I have some questions.

Do you have any plan to support AIP-127 and more AIP spec?

And a bit of advice for `google.protobuf.Any`, maybe we can auto pack and unpack for Any message just like [sisyphus-js](https://github.com/ButterCam/sisyphus-js#well-known-types).
It will be very useful for Improving the development experience.

Thanks.

**Comment by timostamm** [comment]
Thanks for the input, @devkanro I wasn't aware of any JS library that implements REST mappings, nice work!

My focus has been on Twirp and gRPC-web so far. I simply haven't had any jobs yet where AIP-127 was necessary...

It looks like it should be fairly straight-forward to port your [transcoding.ts](https://github.com/ButterCam/sisyphus-js/blob/04666a67f698856f95a5f89943b6ef7760fdc45e/packages/sisyphus-core/src/transcoding.ts) to a protobuf-ts [RpcTransport](https://github.com/timostamm/protobuf-ts/blob/85a3c5ba907169e0edf76d8a55f7d5c42a0051b1/packages/runtime-rpc/src/rpc-transport.ts) (equivalent to your `IRpcImpl`). As far as I see that should be all that is necessary. 

The required reflection information (fields, rpc options) for the transcoding at runtime is already available.  
There already is a [Twirp](https://github.com/timostamm/protobuf-ts/blob/85a3c5ba907169e0edf76d8a55f7d5c42a0051b1/packages/twirp-transport/src/twirp-transport.ts) and a [gRPC-web](https://github.com/timostamm/protobuf-ts/blob/85a3c5ba907169e0edf76d8a55f7d5c42a0051b1/packages/grpcweb-transport/src/grpc-web-transport.ts) implementation, and a "AipTransport" would fit well with that. 

Do you know of any test suites for AIP 127 clients? 

TLDR: No plans for AIP 127 support, but might be trivial to add.

**Comment by timostamm** [comment]
`google.protobuf.Any`: 

We don't generate classes for protobuf messages. They are plain interfaces and don't have a `$type` property where we could look up the type name. 

The advantage is very small code size of the generated code. The disadvantage is that you have to pass the "MessageType" along with instances if you want to perform serialization.

**Comment by devkanro** [comment]
Thanks for your answer, maybe I can create a PR for AIP-127.
